### ISSUE TITLE
Proposal: Remove setAndroidContext, take it at push.activate() instead

### ISSUE THREAD

**Issue by tcard** [ably/ably-java#561]
In order for push activation to work, you need to do this:

```
rest.setAndroidContext(context)
rest.push.activate(useCustomRegistrar)
```

(Which, by the way, [is really poorly documented currently](https://github.com/ably/docs/search?q=setAndroidContext&unscoped_q=setAndroidContext).)

If you do `push.activate` without doing `setAndroidContext` first, [it crashes](https://github.com/ably/ably-java/blob/8e26a834d9de39b3d4c9b822373dce551427e763/android/src/main/java/io/ably/lib/push/Push.java#L50-L51).

So, why not just:

```
rest.push.activate(useCustomRegistrar, androidContext)
```

We use the context exclusively for push, now and for the foreseeable future, so I think it makes sense we take it there.

**Comment by tcard** [comment]
@paddybyers WDYT?

**Comment by paddybyers** [comment]
> We use the context exclusively for push, now and for the foreseeable future, so I think it makes sense we take it there.

Not quite, because it's also used to subscribe for platform network changes (cf RTN20 from memory)
